packages/iris-grid/src/sidebar/VisibilityOrderingBuilder.tsx

```diff
@@ -675,77 +617,333 @@ class VisibilityOrderingBuilder extends Component<
+      name,
+      children,
+      depth: 0,
+      childIndexes,
+    });
+
+    newGroups.push(newGroup);
+
+    model.setColumnHeaderGroups(newGroups);
+    onColumnHeaderGroupChanged(newGroups);
+  }
 
-    return null;
-  });
+  /**
+   * Validates if a header group name is valid and not in use by any header groups or columns
+   * @param groupName The name to validate
+   * @returns True if the name is valid and not in use
+   */
+  validateGroupName(groupName: string): boolean {
+    const { model } = this.props;
+    const columnGroups = model.getColumnHeaderGroups();
+    const { columns } = model;
+    return (
+      DbNameValidator.isValidColumnName(groupName) &&
+      columns.every(({ name }) => name !== groupName) &&
+      columnGroups.every(({ name }) => name !== groupName)
+    );
+  }
 
-  getLastMovableIndex = memoize((model: IrisGridModel, columns: Column[]) => {
-    for (let i = columns.length - 1; i >= 0; i -= 1) {
-      if (model.isColumnMovable(i)) {
-        return i;
+  renderItem = memoize(
+    ({
+      value,
+      clone,
+      childCount,
+      item,
+    }: {
+      value: string;
+      clone: boolean;
+      childCount: number;
+      item: IrisGridTreeItem;
+    }) => {
+      const { onColumnVisibilityChanged } = this.props;
+      const { group, modelIndex, hidden } = item.data;
+
+      return (
+        <>
+          <Button
+            kind="ghost"
+            className="px-1"
+            onClick={() =>
+              onColumnVisibilityChanged(
+                [modelIndex].flat(),
+                hidden
+                  ? VisibilityOrderingBuilder.VISIBILITY_OPTIONS.SHOW
+                  : VisibilityOrderingBuilder.VISIBILITY_OPTIONS.HIDE
+              )
+            }
+            // onKeyDown={visibilityOnKeyDown}
+            icon={hidden ? dhEyeSlash : dhEye}
+            tooltip="Toggle visibility"
+          />
+          <span
+            className="column-name"
+            style={{ flexGrow: 1 }}
+            onClick={e => this.handleItemClick(modelIndex, e)}
+          >
+            {group ? (
+              <VisibilityOrderingGroup
+                group={group}
+                onDelete={this.handleGroupDelete}
+                onColorChange={this.handleGroupColorChange}
+                onNameChange={this.handleGroupNameChange}
+                validateName={this.validateGroupName}
+                isNew={modelIndex === -1}
+              />
+            ) : (
+              value
+            )}
+          </span>
+          <div>
+            {clone && childCount > 1 && (
+              <span className={styles.Count}>{childCount}</span>
+            )}
+            <Tooltip>Drag to re-order</Tooltip>
+            <FontAwesomeIcon icon={vsGripper} />
+          </div>
+        </>
+      );
+    }
+  );
+
+  makeVisibilityOrderingList = memoize(
+    (
+      movedColumns: MoveOperation[],
+      columnHeaderGroups: ColumnHeaderGroup[],
```

Comment:
This is unused in the memoized function

Reply:
I added a comment. It isn't used, but it affects the list when it changes. I think I need to adjust the `getTreeItems` util to take this in instead of grabbing it from the model